danny-avila/LibreChat
# 🎛️ fix: Improve Frontend Practices for Audio Settings

## Summary

Closes #3622 

- Refactored the Dropdown component to use TypeScript in useCallback for better type safety.
- Updated the Speech component to remember the last selected voice, improving user experience.
- Maintained speechTab selections
- Removed await calls inside useCallbacks and relied on updates for dropdowns to enhance performance.
- Updated Dropdown component styles to match the header theme for visual consistency.
- Refactored useTextToSpeech hook to improve voice selection and management.
- Updated useTextToSpeechBrowser, useTextToSpeechEdge, and useTextToSpeechExternal hooks for better state management and error handling.
- Modified the store to use undefined as the default value for the voice atom.
- Updated various components to use common Option type for dropdown options.
- Improved error handling and logging throughout the audio-related components.

Testing:
I thoroughly tested the changes by manually interacting with the audio settings in the application. I verified that voice selection, speech generation, and playback work as expected across different browsers and scenarios. Additionally, I ensured that the UI updates correctly when changing audio settings.

To test these changes:
1. Navigate to the audio settings in the application.
2. Try selecting different voices and verify that the last selected voice is remembered.
3. Generate speech using various engines (browser, edge, external) and ensure playback works correctly.
4. Test error scenarios by intentionally causing failures (e.g., disconnecting internet during speech generation) and verify that error messages are displayed correctly.
